pkgs/native_assets_builder/test_data/treeshaking_native_assets/lib/src/treeshaking_native_assets.dart

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import 'treeshaking_native_assets_bindings.dart' as bindings;
+
+class MyMath {
+  @ResourceIdentifier('add')
```

Comment:
Maybe we should treat all `@Native`s as `@ResourceIdentifier`s? What would be an obvious mapping? we have the `assetId` and the `symbol`.